### ISSUE TITLE
Syndic broken between 2016.3.3 and 2016.11.1

### ISSUE THREAD

**Issue by justindesilets** [saltstack/salt#38827]
### Syndic broken between 2016.3.3 and 2016.11.1
We recently tried to update our salt-master-of-masters to 2016.11.1. We have multiple syndic-masters running 2016.3.3 connected to it. After the update, the syndic functionality was broken. This was done during a scheduled maintenance, so I had to roll back the package and syndic functionality started working as expected. We have however been able to update syndic-masters to 2016.11.1 and they are able to function as expected when talking to the 2016.3.3 master of masters.

### Setup
Our master of masters was running salt-2016.11.1 with multiple 2016.3.3 syndic-masters connected to it.

### Steps to Reproduce Issue
Run a salt master of masters with 2016.11.1 and have a syndic-master running 2016.3.3. The master of masters is unable to receive any return data from the syndic-master.

### Versions Report
Current state of master of masters:
```
Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.1.1503 Core
        machine: x86_64
        release: 3.10.0-327.36.3.el7.x86_64
         system: Linux
        version: CentOS Linux 7.1.1503 Core
```

Current state of one of the syndic-masters:
```Salt Version:
           Salt: 2016.3.3

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: 0.8.1
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.1.1503 Core
        machine: x86_64
        release: 3.10.0-327.36.3.el7.x86_64
         system: Linux
        version: CentOS Linux 7.1.1503 Core
```

**Comment by gtmanfred** [comment]
I have verified that this is a problem with 2016.3.3.  But it seems to work correctly when the syndics are 2016.3.4, could you try that out?

```
[root@saltmaster /]# salt \* test.ping
syndic1:
    True
syndic2:
    True
b152bbc0012a:
    Minion did not return. [No response]
49425bde533a:
    Minion did not return. [No response]
```

but with 2016.3.4

```
[root@saltmaster /]# salt \* test.ping
syndic1:
    True
syndic2:
    True
028e60def084:
    True
a7ba09e84f3e:
    True
[root@saltmaster /]# salt \* test.version
syndic1:
    2016.3.4
syndic2:
    2016.3.4
028e60def084:
    2016.3.4
a7ba09e84f3e:
    2016.3.4
```

I tested using the syndic with docker containers. https://github.com/gtmanfred/salt-docker-demo/tree/syndic

Thanks,
Daniel

**Comment by justindesilets** [comment]
Currently we're in the process of rolling out 2016.11.1 to all hosts so that's our fix that's already scheduled out for all of our sites. I mostly just wanted to bring the issue to light but would have to build a lab to replicate it on my end.

**Comment by gtmanfred** [comment]
Ok, cool. I am going to mark this issue as fixed on 2016.3.4, unless someone comes back with the same error on 2016.3.4

Thanks!
Daniel
